lib/smart_proxy_dynflow_core/log.rb

```diff
@@ -54,13 +54,28 @@ def roll_log
       end
     end
 
+    class ProxyStructuredFormater < ::Dynflow::LoggerAdapters::Formatters::Abstract
+      def call(_severity, _datetime, _prog_name, message)
+        if ::Exception === message
+          @base.exception("Error details", message)
```

Comment:
To count on the fact that the `exception` method might not be present, it should be IMO something like this:

```ruby
          if @base.respond_to?(:exception)
            @base.exception("Error details", message)
            "#{message.message} (#{message.class})"
          else
            "#{message.message} (#{message.class})\n#{message.backtrace.join("\n")}"
          end
```

Reply:
Wait, you just added the method into the wrapper, am I missing something? https://github.com/Dynflow/dynflow/pull/300/files

Reply:
I think we would still miss the `exception` defined in logger when running in separate process, where smart-proxy goodies are not available. Btw. `dynflow-1.1.2` with the required changes as been released: we should probably bump a version requirement here as well.

Reply:
Updated with the inline editor via github.com. Not sure I will be able to update the dependency from here, lemme check.

Reply:
Still not updated fully, could you add something like this, so that we get the backtrace logged properly even when running outside of smart-proxy?:

```ruby
        if ::Exception === message
          subject = "#{message.message} (#{message.class})"
          if @base.respond_to?(:exception)
            @base.exception("Error details", message)
            subject
          else
            "#{subject}\n#{message.backtrace.join("\n")}"
          end
        else
          message
        end
```